src/java/ognl/OgnlRuntime.java

```diff
@@ -899,12 +913,53 @@ public static Object invokeMethod(Object target, Method method, Object[] argsArr
                 }
             }
 
-            result = method.invoke(target, argsArray);
+            result = invokeMethodInJavaSandbox(target, method, argsArray);
         }
 
         return result;
     }
 
+    private static Object invokeMethodInJavaSandbox(Object target, Method method, Object[] argsArray) throws InvocationTargetException, IllegalAccessException {
+        SecurityManager parentSecurityManager = System.getSecurityManager();
+        Policy parentPolicy;
+        try
+        {
+            if (parentSecurityManager != null) {
+                parentSecurityManager.checkPermission(new SecurityPermission("getPolicy"));
+                parentSecurityManager.checkPermission(new SecurityPermission("setPolicy"));
+                parentSecurityManager.checkPermission(new RuntimePermission("setSecurityManager"));
+            }
+            parentPolicy = Policy.getPolicy();
+        } catch (SecurityException ex) {
+            return method.invoke(target, argsArray);
```

Comment:
Not sure if I understand this code, you will invoke a method anyway if there was a `SecurityException`? I would assume it's not allowed to invoke the method, though.

Reply:
I saw in my tests when `Policy.getPolicy()` throws `SecurityException` it means there is already a policy (e.g. my policy applied in previous recursions) that doesn't allow get policy. So we have to honor user higher applied policy and just invoke the method which occurs in and honor higher policy context.